tp/src/wasm_executor/wasm_externals.rs

```diff
@@ -557,6 +567,27 @@ impl<'a> Externals for WasmExternals<'a> {
                     None => Err(ExternalsError::to_trap("No result returned".into())),
                 }
             }
+            LOG => {
+                let log_level: u32 = args.nth(0);
+                let log_ptr: u32 = args.nth(1);
+                let log_string = self.ptr_to_string(log_ptr)?;
+                match log_level {
+                    0 => error!("{}", log_string),
```

Comment:
You could transform the `log_level` int into a `log::Level` value, and then just use `log!`